File: app/src/main/java/com/support/robigroup/ututor/data/network/NetworkHelper.java
package com.support.robigroup.ututor.data.network;

import com.support.robigroup.ututor.features.chat.model.ChatMessage;

import java.io.File;
import java.util.List;

import io.reactivex.Flowable;
import retrofit2.Response;


public interface NetworkHelper {

    Flowable<Response<ChatMessage>> sendAudioMessage(File file);

    Flowable<Response<ChatMessage>> sendImageMessage();

    Flowable<Response<List<ChatMessage>>> getChatMessages(String chatId);

}

File: app/src/main/java/com/support/robigroup/ututor/data/network/NetworkHelper.kt
package com.support.robigroup.ututor.data.network

import com.support.robigroup.ututor.ui.chat.model.ChatMessage

import java.io.File

import io.reactivex.Single
import retrofit2.Response


interface NetworkHelper {

    fun sendAudioMessage(file: File): Single<Response<ChatMessage>>

    fun sendImageTextMessage(messageText: String? = null, file64base: String? = null): Single<Response<ChatMessage>>

    fun getChatMessages(chatId: String): Single<Response<List<ChatMessage>>>?

//    fun downloadAudio(url: String, onProgress: ()->Unit, onFinish: ()->Unit, onError: ()->Unit)
//
//    fun cancelDownload(url: String, onCancelled: ()->Unit)

}
